nomt/tests/fill_and_empty.rs

```diff
@@ -0,0 +1,105 @@
+mod common;
+use common::Test;
+use rand::{prelude::SliceRandom, Rng, SeedableRng};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+fn fill_and_empty(commit_concurrency: usize) {
+    let seed = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .expect("no time?")
+        .as_nanos()
+        .to_le_bytes()[0..16]
+        .try_into()
+        .unwrap();
+
+    let mut rng = rand_pcg::Lcg64Xsh32::from_seed(seed);
+
+    let db_size = 1 << 16;
+    let commit_size = db_size / 16;
+
+    let mut items = std::collections::HashSet::new();
+    while items.len() < db_size as usize {
+        items.insert(rand_key(&mut rng));
+    }
+    let items: Vec<_> = items.into_iter().collect();
+
+    let mut to_delete: Vec<usize> = (0..db_size as usize).collect();
+    to_delete.shuffle(&mut rng);
+
+    let mut t = Test::new_with_params(
+        format!("fill_and_empty_{}", commit_concurrency), // name
+        commit_concurrency,
+        2_000_000, // hashtable_buckets
```

Comment:
this is too many hash-table buckets for 2^16 items. I'd be surprised if we needed more than 20k buckets, so this is going to allocate about 100 times more disk space than needed.

Reply:
You are totally right! I modified the tests to support a custom hashtable_buckets number because I was iterating the tests for multiple db_size values, but anyways 2 million is an exaggeration

Reply:
Anyway, 20k is too low because there is an assertion inside the `BucketAllocator`: `assert!(i < 10000, "hash-table full");`, which makes 20k insufficient. Since the number of pages is not constrained to be a power of two, we are not certain that triangular probing will always find an empty bucket. Therefore, an upper bound is necessary. Perhaps the bound is set too low?

Reply:
10k is too high if anything!